NewHorizons/Utility/Files/ImageUtilities.cs

```diff
@@ -17,7 +19,72 @@ public static class ImageUtilities
         // key is path + applied effects
         private static readonly Dictionary<string, Texture> _textureCache = new();
         public static bool CheckCachedTexture(string key, out Texture existingTexture) => _textureCache.TryGetValue(key, out existingTexture);
-        public static void TrackCachedTexture(string key, Texture texture) => _textureCache.Add(key, texture);
+        public static void TrackCachedTexture(string key, Texture texture)
+        {
+            _textureCache[key] = texture;
+
+            // Save files to thing
+            if (texture is Texture2D texture2D)
+            {
+                try
+                {
+                    var fileName = $"{key.GetHashCode()}.png";
+
+                    var path = Path.Combine(Main.Instance.ModHelper.Manifest.ModFolderPath, "Cache", Main.Instance.CurrentStarSystem, fileName);
+                    Directory.CreateDirectory(Path.GetDirectoryName(path));
+                    File.WriteAllBytes(path, texture2D.EncodeToPNG());
+
+                    // Track the meta data about the image
+                    if (!_cacheTextureLookup.ContainsKey(key))
+                    {
+                        _cacheTextureLookup[key] = new TextureCacheData()
+                        {
+                            useMipmaps = texture.mipmapCount != 1,
+                            linear = false,
+                            wrap = texture.wrapMode == TextureWrapMode.Repeat,
+                            relativePath = fileName
+                        };
+                    }
+                }
+                catch (Exception e)
+                {
+
+                }
+            }
+        }
+
+        public static void LoadCache(string starSystem)
+        {
+            // Read the lookup file for this star system
+            var lookupPath = Path.Combine(Main.Instance.ModHelper.Manifest.ModFolderPath, "Cache", starSystem, "lookup.json");
+            if (File.Exists(lookupPath))
+            {
+                _cacheTextureLookup = JsonConvert.DeserializeObject<Dictionary<string, TextureCacheData>>(File.ReadAllText(lookupPath));
+
+                // Load all the individual files and add them to the cache
+                foreach (var pair in _cacheTextureLookup)
+                {
+                    var key = pair.Key;
+                    var cacheData = pair.Value;
+                    LoadTexture(key, Path.Combine(Main.Instance.ModHelper.Manifest.ModFolderPath, "Cache", starSystem, cacheData.relativePath),
+                        cacheData.useMipmaps, cacheData.wrap, cacheData.linear);
+                }
+            }
+
+            SceneManager.activeSceneChanged += OnSceneChanged;
```

Comment:
think this might get called multiple times since we subscribe to again once per loop